### ISSUE TITLE
Add support for LlamaForSequenceClassification

### ISSUE THREAD

**Issue by ShuaiShao93** [NVIDIA/TensorRT-LLM#3698]
Is it possible to add support for LlamaForSequenceClassification?

I also tried to do it by myself, by changing the output size of lm_head [here](https://github.com/NVIDIA/TensorRT-LLM/blob/f4ddc304f209449993eb15c1a62a0ce8f5c8921a/tensorrt_llm/models/llama/model.py#L412-L418). However, even if I can see correct size (18) in trtllm-build logs, when I actually run inference, both generation logits and context logits are still vocab_size.

If it's not possible to add support, can I at least get some guidance on implementing it by myself?

```
[04/17/2025-19:13:22] [TRT] [V] PLUGIN_V2 LLaMAForCausalLM/lm_head/multiply_collect_L287/multiply_and_lora_L252/_gemm_plugin_L138/PLUGIN_V2_Gemm_0
[04/17/2025-19:13:22] [TRT] [V]     Input 0
[04/17/2025-19:13:22] [TRT] [V]         [1,48000]	num_tokens
[04/17/2025-19:13:22] [TRT] [V]         4096	4096
[04/17/2025-19:13:22] [TRT] [V]     Input 1
[04/17/2025-19:13:22] [TRT] [V]         18	18
[04/17/2025-19:13:22] [TRT] [V]         4096	4096
[04/17/2025-19:13:22] [TRT] [V]     Output 0
[04/17/2025-19:13:22] [TRT] [V]         [1,48000]	num_tokens
[04/17/2025-19:13:22] [TRT] [V]         18	18
```

**Comment by karljang** [comment]
Closing as stale since the TensorRT engine-build path is deprecated in favor of the PyTorch backend. Please open a new issue if the problem persists in the latest version.
